indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractTask.java

```diff
@@ -155,11 +151,6 @@ public String setup(TaskToolbox toolbox) throws Exception
       FileUtils.mkdirp(attemptDir);
       reportsFile = new File(attemptDir, "report.json");
       statusFile = new File(attemptDir, "status.json");
-      InetAddress hostName = InetAddress.getLocalHost();
-      DruidNode node = toolbox.getTaskExecutorNode();
-      toolbox.getTaskActionClient().submit(new UpdateLocationAction(TaskLocation.create(
-          hostName.getHostAddress(), node.getPlaintextPort(), node.getTlsPort(), node.isEnablePlaintextPort()
```

Comment:
Can you mark the `UpdateLocationAction` as `@Deprecated` because it looks like it is no longer used after these changes. Perhaps leave a comment in the javadocs for `UpdateLocationAction` explaining that it is left around to help with rolling upgrades from Druid 25 with mm-less ingestion to Druid 28 with mm-less ingestion. And that it will be removed in a future release